keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/VerificationController.kt

```diff
@@ -33,31 +37,83 @@ class VerificationController(
     @RequestHeader("X-SPINNAKER-USER") user: String,
     @PathVariable("application") application: String,
     @PathVariable("environment") environment: String,
-    @RequestBody status: UpdatedVerificationStatus
+    @RequestBody payload: UpdateVerificationStatusRequest
   ) {
-    require(status.status in listOf(OVERRIDE_PASS, OVERRIDE_FAIL)) {
-      "Only override statuses may be set via this endpoint."
+    if (payload.status !in listOf(OVERRIDE_PASS, OVERRIDE_FAIL)) {
+      throw NotOverrideStatus()
     }
+
     VerificationContext(
       deliveryConfig = deliveryConfigRepository.getByApplication(application),
       environmentName = environment,
-      artifactReference = status.artifactReference,
-      version = status.artifactVersion
+      artifactReference = payload.artifactReference,
+      version = payload.artifactVersion
     ).apply {
       verificationRepository.updateState(
         context = this,
-        verification = verification(status.verificationId),
-        status = status.status,
-        mapOf("overriddenBy" to user, "comment" to status.comment)
+        verification = verification(payload.verificationId),
+        status = payload.status,
+        mapOf("overriddenBy" to user, "comment" to payload.comment)
       )
     }
   }
 
-  data class UpdatedVerificationStatus(
+  data class UpdateVerificationStatusRequest(
     val verificationId: String,
     val artifactReference: String,
     val artifactVersion: String,
     val status: ConstraintStatus,
     val comment: String?
   )
+
+  @ResponseStatus(UNPROCESSABLE_ENTITY)
+  private class NotOverrideStatus :
+    RuntimeException("Only override statuses may be set via this endpoint.")
```

Comment:
Wanted to make this extend `IllegalArgumentException` but then Spring won't let you specify the response code